Enhance workflows list view

VladimirFilonov opened the pull request:
- Introduced `/api/workflows/stats` endpoint for fetching workflows and executions statistics.
- Added UI for rendering Workflows Overview with bar charts displaying execution stats.
- Implemented reusable hook `useWorkflowStats` for fetching stats in React components.
- Extended backend service with methods to retrieve workflow and execution stats.
- Refactor workflows management UI and backend for enhanced search filtering, and cloning capabilities
- Removed obsolete Workflows Overview page.
- Introduced reusable `WorkflowExecutionStatsBar` component.
- Added `/api/workflows/aggs` endpoint for field aggregations.
- Enhanced workflow search API with new params: `page`, `limit`, `query`, `status`, and `createdBy`.
- Implemented workflow cloning functionality via `/api/workflows/{id}/clone`.
- Improved stats endpoint to use 30-second intervals.
- Updated React hooks and components to support new search and filter functionalities.


## Summary

<img width="1441" height="299" alt="image" src="https://github.com/user-attachments/assets/87e37c84-84d9-4a2b-9201-c4c702e89ace" />

--

<img width="1425" height="243" alt="image" src="https://github.com/user-attachments/assets/de2c2f3c-f642-4dde-b3a4-6ee9c8c806ac" />
<img width="1420" height="394" alt="image" src="https://github.com/user-attachments/assets/b2377a3d-25b3-49e4-ac5c-096dc5c3475e" />



### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

VladimirFilonov committed:
Add workflows overview and statistics APIs/UI

- Introduced `/api/workflows/stats` endpoint for fetching workflows and executions statistics.
- Added UI for rendering Workflows Overview with bar charts displaying execution stats.
- Implemented reusable hook `useWorkflowStats` for fetching stats in React components.
- Extended backend service with methods to retrieve workflow and execution stats.

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

VladimirFilonov committed:
Refactor workflows management UI and backend for enhanced search, filtering, and cloning capabilities

- Removed obsolete Workflows Overview page.
- Introduced reusable `WorkflowExecutionStatsBar` component.
- Added `/api/workflows/aggs` endpoint for field aggregations.
- Enhanced workflow search API with new params: `page`, `limit`, `query`, `status`, and `createdBy`.
- Implemented workflow cloning functionality via `/api/workflows/{id}/clone`.
- Improved stats endpoint to use 30-second intervals.
- Updated React hooks and components to support new search and filter functionalities.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/329848)
* Commit: ed56a7f21994b206bda05146511a0d5c770d6136

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/329848#0198aa25-0c6d-41ca-ab5e-a93ee7b26e8f)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329848","number":329848,"commit":"ed56a7f21994b206bda05146511a0d5c770d6136"}],"number":329848}
buildkite-pr-comment-->

VladimirFilonov committed:
Refactor workflows management pagination and filtering

- Replaced `offset` with `page` for pagination in DTOs and APIs.
- Updated UI components to use `FieldValueOptionType` for filter values.
- Improved workflows list table interaction and added additional execution statuses.
- Adjusted default and available table page size options.
- Enhanced mutation keys for workflow actions.

Kiryous commented on src/platform/plugins/shared/workflows_management/server/workflows_management/workflows_management_service.ts:
```diff
@@ -100,12 +108,39 @@ export class WorkflowsService {
+
+    const filterQuery = filters.join(' AND ');
+
     const response = await savedObjectsClient.find<WorkflowSavedObjectAttributes>({
```
Noticed a few bugs:
1. pagination seems to work incorrectly: changing page doesn't change results 
2. partial name in search doesn't return, e.g. ("slac" will not return workflows with names "slack 1", "slack 2")

Video:

https://github.com/user-attachments/assets/4b3cbb9a-bed4-4e43-86e1-54cb8dd89511

Kiryous reviewed (changes requested):
Noticed a few bugs ,please have a look:

VladimirFilonov committed:
Fix workflows pagination logic

- Adjusted search state initialization to start `page` from 1.
- Corrected `showStart` and `showEnd` calculations for accurate item ranges.
- Updated table pagination handling to ensure proper page indices.

VladimirFilonov committed:
Enhance filtering and pagination for workflows management

- Updated query filtering logic to allow partial matches.
- Replaced `offset` with `page` in pagination response for consistency.

VladimirFilonov committed:
Refactor workflows management codebase

- Introduced shared translations for improved i18n support.
- Standardized imports for type and default exports across files.
- Replaced inline text with i18n translations in workflow search field.
- Centralized execution stats interval into a constant.
- Adjusted aggregated response keys for better readability (`day` -> `interval`).
- Simplified `WorkflowAggsDto` type definition.

VladimirFilonov committed:
Refactor workflows management to replace `status` field with `enabled`

- Migrated `status` to a boolean `enabled` field across UI, backend, and types.
- Updated filtering, search, and stats APIs to handle the new `enabled` field.
- Removed the `WorkflowStatus` enum in favor of simplification.
- Improved field aggregation logic for better handling of data types.

VladimirFilonov committed:
Update `enabled` field to support arrays in workflows filtering

- Replaced single boolean `enabled` with an array of booleans in types and API.
- Updated workflows filtering logic and request handling to align with the new structure.

VladimirFilonov committed:
Add `UpdateWorkflowCommandSchema` and extend workflows management API

- Introduced `UpdateWorkflowCommandSchema` for handling workflow updates.
- Updated API routes to support workflow updates with partial schemas.
- Added utility method `updateYAMLFields` for processing YAML updates.
- Enhanced workflow cloning to include localized naming logic.
- Refined workflow filtering logic to support state queries.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

VladimirFilonov committed:
Update `status` field to `enabled` in mappings

- Replaced `status` field with `enabled` in mappings.

VladimirFilonov committed:
Enhance workflow cloning functionality and align fields with API updates

- Updated `cloneWorkflow` method to include `request` parameter for improved context handling.
- Changed `status` field to `enabled` and refined related logic in DTOs and tests.
- Replaced `offset` with `page` in pagination for consistency.
- Adjusted optional types in query parameters for better flexibility in workflows API.

VladimirFilonov committed:
Refactor workflow execution stats handling

- Changed `date_histogram` to `auto_date_histogram` with dynamic bucket computation.
- Updated UI to remove interval reference from axis title.
- Replaced hardcoded 30-second interval with configurable bucket size.

VladimirFilonov committed:
Refactor workflows management: replace `status` with `enabled` and update mappings

- Migrated `status` to `enabled` across UI, backend, and types.
- Updated mappings to add `enabled` as a boolean field and deprecated `status`.
- Adjusted related logic in workflows processing for consistency.

VladimirFilonov committed:
Refine `enabled` field type definition in workflow mappings

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/331065)
* Commit: 24c35c883764cb905bd8f849a670490f34a9cec8

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/331065#0198c696-1721-4d31-adba-34b9ce6591de)

### History
* :broken_heart: [Build #331046](https://buildkite.com/elastic/kibana-pull-request/builds/331046) failed 8cce505e71558829c2db964317e6d41919a1be97
* :broken_heart: [Build #330799](https://buildkite.com/elastic/kibana-pull-request/builds/330799) failed a5f6c24d51ec8ed7a5d9088c658e19401e27b608
* :broken_heart: [Build #330666](https://buildkite.com/elastic/kibana-pull-request/builds/330666) failed 0cc35a684bcc1e128c6ca1375587f8706769addc
* :broken_heart: [Build #330472](https://buildkite.com/elastic/kibana-pull-request/builds/330472) failed 21676c19a614c7dcc6a847ce888f63c52a843454
* :broken_heart: [Build #330451](https://buildkite.com/elastic/kibana-pull-request/builds/330451) failed fd0e5f31b89ed7584db8d9f698ee63112c34e7d0

cc @VladimirFilonov

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331065","number":331065,"commit":"24c35c883764cb905bd8f849a670490f34a9cec8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331046","number":331046,"commit":"8cce505e71558829c2db964317e6d41919a1be97"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330799","number":330799,"commit":"a5f6c24d51ec8ed7a5d9088c658e19401e27b608"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330666","number":330666,"commit":"0cc35a684bcc1e128c6ca1375587f8706769addc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330472","number":330472,"commit":"21676c19a614c7dcc6a847ce888f63c52a843454"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330451","number":330451,"commit":"fd0e5f31b89ed7584db8d9f698ee63112c34e7d0"}],"number":331065}
buildkite-pr-comment-->

VladimirFilonov committed:
Update workflows search API calls to include pagination parameters

- Added `limit`, `page`, and `query` parameters in `body` for `/api/workflows/search` requests.
- Updated related tests to verify correct API call structure.
- Adjusted `mockSavedObjectsClient.find` to include `page` parameter for consistency.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/331070)
* Commit: 8270721a4fc999ee8d3ae11cadae36f736b72c82

### Failed CI Steps
* [Checks](https://buildkite.com/elastic/kibana-pull-request/builds/331070#0198c6a7-fb9b-4bbc-a15c-94eb519a49ee)

### History
* :broken_heart: [Build #331065](https://buildkite.com/elastic/kibana-pull-request/builds/331065) failed 24c35c883764cb905bd8f849a670490f34a9cec8
* :broken_heart: [Build #331046](https://buildkite.com/elastic/kibana-pull-request/builds/331046) failed 8cce505e71558829c2db964317e6d41919a1be97
* :broken_heart: [Build #330799](https://buildkite.com/elastic/kibana-pull-request/builds/330799) failed a5f6c24d51ec8ed7a5d9088c658e19401e27b608
* :broken_heart: [Build #330666](https://buildkite.com/elastic/kibana-pull-request/builds/330666) failed 0cc35a684bcc1e128c6ca1375587f8706769addc
* :broken_heart: [Build #330472](https://buildkite.com/elastic/kibana-pull-request/builds/330472) failed 21676c19a614c7dcc6a847ce888f63c52a843454
* :broken_heart: [Build #330451](https://buildkite.com/elastic/kibana-pull-request/builds/330451) failed fd0e5f31b89ed7584db8d9f698ee63112c34e7d0

cc @VladimirFilonov

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331070","number":331070,"commit":"8270721a4fc999ee8d3ae11cadae36f736b72c82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331065","number":331065,"commit":"24c35c883764cb905bd8f849a670490f34a9cec8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331046","number":331046,"commit":"8cce505e71558829c2db964317e6d41919a1be97"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330799","number":330799,"commit":"a5f6c24d51ec8ed7a5d9088c658e19401e27b608"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330666","number":330666,"commit":"0cc35a684bcc1e128c6ca1375587f8706769addc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330472","number":330472,"commit":"21676c19a614c7dcc6a847ce888f63c52a843454"}],"number":331070}
buildkite-pr-comment-->

VladimirFilonov committed:
Update `status` to `enabled` and revise mapping type

- Replaced `status` field with `enabled` in `current_fields.json`.
- Updated `enabled` field type in `current_mappings.json` from `true` to `boolean`.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/check_mappings_update --fix'

Kiryous reviewed (approved):
Awesome work! Left one comment tho

VladimirFilonov committed:
Fix workflows status mapping and type handling

- Corrected `disabled` workflow mapping in status response.
- Updated type conversion for boolean keys to ensure accuracy.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/331144)
* Commit: 2b01a01c238e2f813031d87ae0a39cf35200757f

### Failed CI Steps
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/331144#0198c761-e93d-461c-b981-379f7c114021)
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/331144#0198c784-9990-476d-b952-cdf771e699df)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/331144#0198c761-e93d-461c-b981-379f7c114021) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/331144/jobs/0198c761-e93d-461c-b981-379f7c114021/artifacts/0198c784-887b-4c2c-81e5-0b834e86e9c0) FTR Configs #<span></span>62 / Reporting APIs Default reporting_user role able to generate CSV report
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/331144#0198c784-9990-476d-b952-cdf771e699df) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/331144/jobs/0198c784-9990-476d-b952-cdf771e699df/artifacts/0198c78b-6476-4c98-8aca-4f32d1b62443) FTR Configs #<span></span>62 / Reporting APIs Default reporting_user role able to generate CSV report

### History
* :broken_heart: [Build #331070](https://buildkite.com/elastic/kibana-pull-request/builds/331070) failed 8270721a4fc999ee8d3ae11cadae36f736b72c82
* :broken_heart: [Build #331065](https://buildkite.com/elastic/kibana-pull-request/builds/331065) failed 24c35c883764cb905bd8f849a670490f34a9cec8
* :broken_heart: [Build #331046](https://buildkite.com/elastic/kibana-pull-request/builds/331046) failed 8cce505e71558829c2db964317e6d41919a1be97
* :broken_heart: [Build #330799](https://buildkite.com/elastic/kibana-pull-request/builds/330799) failed a5f6c24d51ec8ed7a5d9088c658e19401e27b608
* :broken_heart: [Build #330666](https://buildkite.com/elastic/kibana-pull-request/builds/330666) failed 0cc35a684bcc1e128c6ca1375587f8706769addc

cc @VladimirFilonov

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331144","number":331144,"commit":"2b01a01c238e2f813031d87ae0a39cf35200757f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331070","number":331070,"commit":"8270721a4fc999ee8d3ae11cadae36f736b72c82"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331065","number":331065,"commit":"24c35c883764cb905bd8f849a670490f34a9cec8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/331046","number":331046,"commit":"8cce505e71558829c2db964317e6d41919a1be97"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330799","number":330799,"commit":"a5f6c24d51ec8ed7a5d9088c658e19401e27b608"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330666","number":330666,"commit":"0cc35a684bcc1e128c6ca1375587f8706769addc"}],"number":331144}
buildkite-pr-comment-->

VladimirFilonov committed:
Update workflow cloning and management to handle space awareness

- Enhanced `getWorkflow` and `cloneWorkflow` methods to include `spaceId` for multi-space support.
- Updated API routes and service calls to align with space-aware workflows.
- Refined error handling and context passing in cloning logic.
